### ISSUE TITLE
GraphicalLassoCV Example

### ISSUE THREAD

**Issue by AntoBr96** [skfolio/skfolio#63]
#### GraphicalLassoCV Example

Is it possible to have an example where the precision matrix obtained with GraphicalLassoCV is employed?

**Comment by HugoDelatte** [comment]
Do you have more details or literature references in mind? 

All precision matrices from scikit-learn covariance estimators are computed using the same function: `scipy.linalg.pinvh`.
Also, it has been shown that solving MVO using conic optimization is superior than Quadratic Optimization models.
So in `skfolio`, we don't use the precision matrix but the Cholesky decomposition.

The regularization benefit you obtain using the `GraphicalLassoCV` estimator of the covariance will naturally be passed to the Cholesky decomposition.

**Comment by AntoBr96** [comment]
Hi Hugo! Thank you for your quick answer. My question is more about the possibility of including an example where the GraphicalLassoCV is used. It returns a sparse inverse covariance, not a covariance like the other methods, so it would be useful to show how these different outputs are used.

**Comment by AntoBr96** [comment]
That is exactly my doubt... Why not using the sparse inverse covariance? How to directly use it?

**Comment by HugoDelatte** [comment]
We don't use directly the precision matrix because the problem is solved using Second-order cone programming which needs the Cholesky decomposition. For examples, have a look at https://docs.mosek.com/portfolio-cookbook/markowitz.html#conic-formulation. Also, SOCP makes it possible to solve multi-objectives optimizations with the other risk measures.

The question is: do we keep the sparsity benefit in the current implementation, and if not, how can we sort that problem? 
It will require a bit of reasearch as I don't have the answer.

**Comment by albystack** [comment]
Hi @HugoDelatte , I’d be interested in working on this. I’ve started looking into how the sparse precision matrix from GraphicalLassoCV interacts with skfolio’s current covariance/Cholesky SOCP formulation. Would you be happy for me to investigate whether the sparsity benefit is preserved and, depending on the findings, either add an example/documentation or propose an implementation change if there is a meaningful improvement?

**Comment by HugoDelatte** [comment]
@albystack, thanks and yes this is still an open problem. I would suggest to work from the branch `feature/characteristics-factor-model` that will be released into v1 pretty soon (https://github.com/skfolio/skfolio/pull/253) and contains changes around Cholesky and SOCP formulation.

**Comment by albystack** [comment]
Thanks Hugo, I’ve started implementing an approach on top of `feature/characteristics-factor-model`. I still need to run the full test suite and validate the numerical behaviour/benchmarks, but once that’s done I’ll open a PR for review!

**Comment by albystack** [comment]
I’ve opened a draft PR for this in #258.

I also investigated whether the sparse precision structure could be exploited directly in the SOCP formulation. An exact auxiliary formulation looks promising when the precision matrix is explicitly guaranteed to be the exact inverse of the covariance, but that isn’t a contract provided by the current GraphicalLassoCV outputs. Using precision_ directly could therefore slightly change the covariance-defined optimisation problem.

For this PR I’ve kept the solver behaviour unchanged and focused on the example and clarifying the distinction between precision sparsity, covariance sparsity and Cholesky fill-in. Would appreciate your thoughts on whether you agree with that scope!

**Comment by HugoDelatte** [comment]
@albystack thanks for the detailed analysis.

I also thought about exposing the precision matrix in the public API and after reviewing your analysis I agree it's probably not worth it. The only real benefit would be a sparser variance SOC constraint, but:

* For moderate universe (~1000 assets), the dense Cholesky SOC is fast and rarely the bottleneck. Both formulations give the same solution, so there is likely nothing to gain in accuracy or convergence. The precision version also adds n extra variables and n equality constraints, so on small problems it may even be slower
* To keep the constraint sparse, the Cholesky factor of the precision matrix needs to stay sparse. That means a fill-reducing ordering, so a new dependency (scikit-sparse/CHOLMOD) and as you noted, fill-in can remove most of the sparsity
* Above a few thousand assets (where sparsity would actually benefits) `GraphicalLassoCV` is likely the bottleneck anyway (cubic per iteration)
* For large universes we already have the factor model path: `ReturnDistribution.covariance_sqrt` uses the low-rank plus diagonal structure from `FactorModel`, which reduces the SOC to n x k plus a diagonal (usually a much bigger reduction vs graph sparsity)

If someone comes with a real use case (large universe, truly sparse dependence structure, no factor structure, benchmarks showing the speedup holds after fill-in), the way to do it would be a sibling of `CovarianceSqrt` consumed internally by the variance SOC.

I'll review your tutorial in the coming days.

**Comment by albystack** [comment]
Hey @HugoDelatte, I noticed #258 was automatically closed when feature/characteristics-factor-model was deleted after #253 was merged. My branch/changes are still intact, so I’m planning to rebase the GraphicalLassoCV example onto the current main and open a new PR targeting main. Just wanted to check that’s still the preferred direction.

**Comment by HugoDelatte** [comment]
Ah, I see what happened. The PR branch was deleted after it was merged, and since yours was based on it, GitHub closed your PR as well.
Yes you can rebase to main and reopen your PR, thank you.

**Comment by albystack** [comment]
Thanks Hugo. Since #258 was closed when the feature branch was deleted, I rebased the work onto `main` and opened replacement Draft PR #264:
https://github.com/skfolio/skfolio/pull/264

The PR adds the requested `GraphicalLassoCV` tutorial. It explains `covariance_`, `precision_`, partial correlations, conditional independence, and the distinction between statistical sparsity and computational sparsity.

The investigation found that using `precision_` directly would not be safe under the current API because the returned covariance and precision matrices are not guaranteed to be exact numerical inverses. The tutorial therefore keeps `covariance_` as the authoritative risk representation and makes no unsupported solver-performance claim.

I would appreciate your review of the explanation and scope.
